src/classes/HiFiCommunicator.ts

```diff
@@ -53,6 +96,70 @@ export enum HiFiUserDataStreamingScopes {
     All = "all"
 };
 
+export interface ConnectionRetryAndTimeoutConfig {
+  /**
+   * Whether or not to automatically retry initial connection attempts.
+   * When this is set to true, if the first attempt to connect to
+   * the High Fidelity servers fails, we will automatically retry the connection.
+   * While the connection is being attempted, the overall HiFiCommunicator state (as queried via
+   * `getConnectionState()`) will be "Connecting".
+   * After the desired amount of time has passed, if a connection has not been established, we will stop
+   * trying to reconnect and the connection state will change to 'Failed'.
+   * By default, connections are not retried. However, when we start a connection, we do always
+   * trigger a state change to HiFiConnectionStates.Connecting when we start the
+   * connection process, regardless of the setting of this value.
+   *
+   */
+  autoRetryInitialConnection?: boolean;
+  /**
+   * The total amount of time (in seconds) to keep retrying the initial
+   * connection before giving up completely. If `autoRetryInitialConnection`
+   * is set to `true`, this defaults to 60 seconds.
+   * TODO: Don't let this be negative
+   */
+  maxSecondsToSpendRetryingInitialConnection?: number;
+
+  /**
+   * Whether or not to automatically attempt to reconnect if an existing
+   * connection is disconnected. When this is set to true, we will attempt
+   * to reconnect if any disconnect from any cause occurs.
+   * By default, reconnections are not automatically attempted.
+   * NOTE: The retrying that happens when this is set to `true` does not currently take into account
+   * the reason WHY a connection was disconnected. This means that if this is
+   * set to true, a connection that is disconnected via a purposeful server-side
+   * action (e.g. a "kick") will be automatically reconnected. (However, connections
```

Comment:
Are we creating a bug? What overall system behavior are we/users expecting for kick?

Reply:
Excellent question! This came up back when we were initially talking about autoreconnects. I would expect that people are thinking that "kick" would not trigger a client-side autoreconnect, and we https://highfidelity.atlassian.net/browse/HIFI-630 was filed for having the server explicitly tell the client when a kick was happening so that it could trap kicks. For now I think it's probably acceptable to document that "autoreconnects and kicks are incompatible" in the spirit of making incremental changes. I don't think it _quite_ qualifies as a "bug", but it does need to be aggressively documented for sure.